File: MMORPG_2/Assets/Script/Data/Proto/CreateLuaProto/BagListRequestProto.lua
--背包列表(请求)
BagListRequestProto = { ProtoCode = 10301 }

--这句是重定义元表的索引，就是说有了这句，这个才是一个类
BagListRequestProto.__index = BagListRequestProto;

function BagListRequestProto.New()
    local self = { }; --初始化self
    setmetatable(self, BagListRequestProto); --将self的元表设定为Class
    return self;
end


--发送协议
function BagListRequestProto.SendProto(proto)

    local ms = CS.LuaHelper.Instance:CreateMemoryStream();
    ms:WriteUShort(proto.ProtoCode);


    CS.LuaHelper.Instance:SendProto(ms:ToArray());
    ms:Dispose();
end


--解析协议
function BagListRequestProto.GetProto(buffer)

    local proto = BagListRequestProto.New(); --实例化一个协议对象
    local ms = CS.LuaHelper.Instance:CreateMemoryStream(buffer);


    ms:Dispose();
    return proto;
end

File: GameServer/GameServerApp/GameServerApp/Proto/CreateProto/BagListRequestProto.cs
//===================================================
//作    者：xxx
//创建时间：2019-05-18 15:13:23
//备    注：
//===================================================
using System.Collections;
using System.Collections.Generic;
using System;

/// <summary> 背包列表(请求) </summary>
public struct BagListRequestProto : IProto
{
    public ushort ProtoCode { get { return 10301; } }


    public byte[] ToArray()
    {
        using (MMO_MemoryStream ms = new MMO_MemoryStream())
        {
            ms.WriteUShort(ProtoCode);
            return ms.ToArray();
        }
    }

    public static BagListRequestProto GetProto(byte[] buffer)
    {
        BagListRequestProto proto = new BagListRequestProto();
        using (MMO_MemoryStream ms = new MMO_MemoryStream(buffer))
        {
        }
        return proto;
    }
}